packages/paste-core/components/form/src/FormSection.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import {Box, safelySpreadBoxProps} from '@twilio-paste/box';
+import type {BoxProps} from '@twilio-paste/box';
+import {useUID} from '@twilio-paste/uid-library';
+
+import {FormSectionContext} from './FormSectionContext';
+
+export interface FormSectionProps extends Omit<React.ComponentPropsWithRef<'fieldset'>, 'children'> {
+  element?: BoxProps['element'];
+  children: React.ReactNode;
+}
+
+export const FormSection = React.forwardRef<HTMLFieldSetElement, FormSectionProps>(
+  ({children, element = 'FORM_SECTION', id: idProp, ...props}, ref) => {
+    const id = idProp ? idProp : useUID();
+
+    return (
+      <FormSectionContext.Provider value={{id}}>
+        <Box
+          as="fieldset"
+          border="none"
+          margin="space0"
+          padding="space0"
+          marginBottom="space60"
+          ref={ref}
+          element={element}
+          display="flex"
+          flexDirection="column"
+          rowGap="space80"
+          _last={{marginBottom: 'space0'}}
+          {...safelySpreadBoxProps(props)}
+          aria-describedby={`${id}-section-description`}
```

Comment:
thought: would this break our i18n rules for no strings? Or since it's used more like a key here will it not be something actually seen by a user?